examples/gallery/symbols/patterns.py

```diff
@@ -0,0 +1,76 @@
+r"""
+Bit and Hachure Patterns
+------------------------
+
+PyGMT allows using bit or hachure patterns via the ``fill`` parameter
+or similar parameters:
+
+- Symbols and polygons: :meth:`pygmt.Figure.plot` via ``fill``
+- Histogram bars or sectors: :meth:`pygmt.Figure.histogram` or
+  :meth:`pygmt.Figure.rose` via ``fill``
+- Anomalies: :meth:`pygmt.Figure.wiggle` via ``fillpositive``
+  and ``fillnegative``
+- Land and water masses: :meth:`pygmt.Figure.coast` via ``land``
+  and ``water``
+- Focal mechanisms: :meth:`pygmt.Figure.meca` via ``G`` and ``E``
+
+The required argument has the following form:
+
+**P**\ |**p**\ *pattern*\ [**+b**\ *color*\ ][**+f**\ *color*\ ][**+r**\ *dpi*]
```

Comment:
I wonder if we should define a class for GMT patterns, e.g.,
```
class GMTPattern:
    def __init__(self, id, reversed=False, bgcolor="white", fgcolor="black", dpi=1200):
        ...
```

Reply:
Additional to the list at the beginning of this example, there is also the possibility to add a pattern via the `dcw` and `frame` parameters. The syntax seems to be difficult, as it seems to be something like **+g** **p|P** _resolution_ / _pattern_. The functionality seems to be limited, as it seems that no foreground and background colors could be given (please correct me if I am wrong, so far I did not figure out how to set colors here). 

_Example code_
```python
import pygmt

fig = pygmt.Figure()

with fig.subplot(nrows=1, ncols=2, figsize=("10c", "5c")):

    with fig.set_panel(panel=0):
        fig.basemap(
            region=[-12, 50, 30, 70],
            projection="M?",
            frame="afg",
        )
        fig.coast(
            land="lightbrown",
            water="lightblue",
            shorelines="1/0.5p,black",
            borders="1/0.25p,red",
            dcw="=EU+gp200/8",  # here resolution/pattern 
        )        

    with fig.set_panel(panel=1):
        fig.basemap(
            region=[0, 10, 0, 10],
            projection="X?",
            frame=["afg", "+gp400/8"],  # here resolution/pattern
        )

fig.show()
# fig.savefig(fname="pattern_dcw_frame.png")
```

_Output figure_
![pattern_dcw_frame](https://user-images.githubusercontent.com/94163266/215422438-d8d8675e-b659-4e29-b11d-e6f916a105f5.png)

Reply:
> The syntax seems to be difficult, as it seems to be something like **+g** **p|P** _resolution_ / _pattern_.

This is the old syntax of patterns for GMT 4. The full old syntax is **-Gp**_dpi_/_pattern_[**:B**_color_[**F**_color_]]. I think the old syntax is still supported in GMT5 and GMT6 (backward compatibility), but it's no longer documented.

> The functionality seems to be limited, as it seems that no foreground and background colors could be given.

As I understand it, it's impossible to specify foreground/background colors and dpi using the new syntax in `coast -E`.

Ping @PaulWessel for comments on this.

Reply:
> I wonder if we should define a class for GMT patterns, e.g.,
> 
> ```
> class GMTPattern:
>     def __init__(self, id, reversed=False, bgcolor="white", fgcolor="black", dpi=1200):
>         ...
> ```

Maybe we should we open a separate issue for this?